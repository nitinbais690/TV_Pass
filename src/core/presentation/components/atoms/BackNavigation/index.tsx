import React from 'react';
import { View } from 'react-native';
import { backNavigationStyle } from './styles';
import { Text } from 'react-native-elements';
import BackArrow from 'assets/images/back_arrow.svg';
import { useAppPreferencesState } from 'utils/AppPreferencesContext';
import { useSafeArea } from 'react-native-safe-area-context';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { AUTOMATION_TEST_ID } from 'core/presentation/automation-ids';

export default function BackNavigation(props: BackNavigationProps) {
    const prefs = useAppPreferencesState();
    const { appTheme } = prefs;
    let { appColors } = appTheme && appTheme(prefs);

    const styles = backNavigationStyle(appColors, props.isFullScreen, useSafeArea());
    return (
        <View style={[styles.containerStyle, props.containerStyle]}>
            <TouchableOpacity
                style={{ padding: 10 }}
                onPress={props.onPress}
                testID={AUTOMATION_TEST_ID.BACK_NAVIGATION}
                accessibilityLabel={AUTOMATION_TEST_ID.BACK_NAVIGATION}>
                <BackArrow />
            </TouchableOpacity>
            {props.navigationTitle && <Text style={styles.navigationTitleStyle}>{props.navigationTitle}</Text>}
        </View>
    );
}

interface BackNavigationProps {
    navigationTitle?: string;
    containerStyle?: {};
    onPress: () => void;
    isFullScreen: boolean;
}
