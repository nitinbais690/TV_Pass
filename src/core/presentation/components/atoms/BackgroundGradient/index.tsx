import React from 'react';
import { View, StyleSheet, Platform } from 'react-native';
import { BottomTabBarHeightContext } from '@react-navigation/bottom-tabs';
import { HeaderHeightContext } from '@react-navigation/stack';
import { useHeaderTabBarHeight } from 'core/presentation/components/molecules/HeaderTabBar';
import LinearGradient from 'react-native-linear-gradient';
import AHAGreyLogo from 'assets/images/aha_grey_logo.svg';

import { SvgCss } from 'react-native-svg';
import { BackgroundGradientProps } from 'core/data/models/BackgroundGradientProps';
import { xml } from './xml/BackgroundGradientXml';
import { scale } from 'qp-common-ui';

const BackgroundGradient = (props: React.PropsWithChildren<BackgroundGradientProps>): JSX.Element => {
    const headerHeight = React.useContext(HeaderHeightContext) || 0;
    const headerTabBarHeight = useHeaderTabBarHeight();
    const insetHeader = props.insetHeader === undefined ? true : props.insetHeader;
    const tabBarHeight = React.useContext(BottomTabBarHeightContext) || 0;

    const styles = StyleSheet.create({
        container: {
            flex: 1,
        },
        gradientViewStyle: {
            flex: 1,
            width: '100%',
            height: '100%',
        },
        childContainer: {
            position: 'absolute',
            width: '100%',
            height: '100%',
            top: 0,
            bottom: 0,
            left: 0,
            right: 0,
            paddingTop:
                insetHeader === true
                    ? props.headerType === 'HeaderTab'
                        ? headerTabBarHeight
                        : headerHeight
                    : undefined,
            paddingBottom: props.insetTabBar === true ? tabBarHeight : undefined,
        },
        greyLogoStyle: {
            width: scale(42),
        },
        greyLogoPosition: {
            position: 'absolute',
            right: 50,
            top: 12,
        },
    });

    const gradientVariantColorsTV = {
        primaryVariant1: '#532C1E',
        primaryVariant2: '#24272B',
        primaryVariant3: '#101211',
        primaryVariant4: '#3C2924',
    };

    return (
        <View style={styles.container} {...props}>
            {!Platform.isTV && (
                <SvgCss xml={xml} width={styles.gradientViewStyle.width} height={styles.gradientViewStyle.height} />
            )}
            {!Platform.isTV && <View style={[styles.childContainer, props.childContainerStyle]}>{props.children}</View>}
            {Platform.isTV && (
                <LinearGradient
                    style={styles.gradientViewStyle}
                    colors={[
                        gradientVariantColorsTV.primaryVariant1,
                        gradientVariantColorsTV.primaryVariant2,
                        gradientVariantColorsTV.primaryVariant3,
                        gradientVariantColorsTV.primaryVariant4,
                    ]}
                    locations={[0, 0.035, 0.53, 1]}
                    useAngle={true}
                    angle={140}>
                    <AHAGreyLogo width={styles.greyLogoStyle.width} style={styles.greyLogoPosition} />
                    <View style={[styles.childContainer, props.childContainerStyle]}>{props.children}</View>
                </LinearGradient>
            )}
        </View>
    );
};

export default BackgroundGradient;
