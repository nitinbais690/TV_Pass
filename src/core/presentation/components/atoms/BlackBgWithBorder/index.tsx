import useAppColors from 'core/presentation/hooks/use-app-colors';
import React from 'react';
import { View, Text } from 'react-native';
import { styles } from '../BlackBgWithBorder/styles';

export default function BlackBgWithBorder(props: BlackBgWithBorderProps) {
    const appColors = useAppColors();
    const style = styles(appColors);
    return (
        <View style={style.containerStyle}>
            <Text style={style.titleStyle}>{props.text}</Text>
        </View>
    );
}

interface BlackBgWithBorderProps {
    text: string;
    style?: {};
}
