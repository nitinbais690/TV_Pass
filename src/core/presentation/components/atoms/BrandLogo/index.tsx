import React from 'react';
import AhaLogo from 'assets/images/aha-logo.svg';

export default function BrandLogo(props: { width: number; height: number }) {
    return <AhaLogo width={props.width} height={props.height} />;
}
