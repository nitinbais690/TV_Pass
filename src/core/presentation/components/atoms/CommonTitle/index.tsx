import useAppColors from 'core/presentation/hooks/use-app-colors';
import { appFontStyle } from 'core/styles/AppStyles';
import React from 'react';
import { View, Text } from 'react-native';
import { headerTextStyle } from './styles';
import AhaLogo from 'assets/images/aha_mini_logo.svg';

export default function CommonTitle(props: HeaderTextprops) {
    var value: string = props.text;
    const appColors = useAppColors();
    if (props.showThemedDot && value) {
        value = value.endsWith('.') ? value.slice(0, -1) : value;
        return (
            <View style={props.style}>
                <Text style={[appFontStyle.header1, headerTextStyle(appColors).textStyle]}>
                    {value}
                    <Text style={[appFontStyle.header1, headerTextStyle(appColors).dotText]}>.</Text>
                </Text>
            </View>
        );
    } else {
        return (
            <Text style={[appFontStyle.header1, headerTextStyle(appColors).textStyle, props.style]}>
                {props.text} {props.showAhaLogo && <AhaLogo />}
            </Text>
        );
    }
}

interface HeaderTextprops {
    text: string;
    showThemedDot: boolean;
    showAhaLogo?: boolean;
    style?: {};
}
