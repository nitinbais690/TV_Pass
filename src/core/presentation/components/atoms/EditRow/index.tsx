import React from 'react';
import { Text, TouchableHighlight, View } from 'react-native';
import { profileListItemStyle } from './styles';
import { appFlexStyles } from 'core/styles/FlexStyles';
import { useLocalization } from 'contexts/LocalizationContext';
import EditIcon from 'assets/images/ic_edit.svg';
import useAppColors from 'core/presentation/hooks/use-app-colors';
import { AUTOMATION_TEST_ID } from 'core/presentation/automation-ids';

const EditRow = (props: EditRowProps) => {
    const styles = profileListItemStyle(useAppColors());
    const { strings } = useLocalization();

    return (
        <TouchableHighlight
            onPress={props.onPress}
            testID={AUTOMATION_TEST_ID.EDIT + props.testID}
            accessibilityLabel={AUTOMATION_TEST_ID.EDIT + props.accessibilityLabel}>
            <View style={[appFlexStyles.flexRow, appFlexStyles.rowVerticalAlignCenter]}>
                <EditIcon style={styles.icon} />
                {props.showText && (
                    <Text style={styles.editText}>{props.editText ? props.editText : strings.edit}</Text>
                )}
            </View>
        </TouchableHighlight>
    );
};

interface EditRowProps {
    showText: boolean;
    onPress: () => void;
    editText?: string;
    testID?: string;
    accessibilityLabel?: string;
}

export default EditRow;
