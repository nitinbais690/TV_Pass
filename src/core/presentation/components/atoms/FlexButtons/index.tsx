import React from 'react';
import { View } from 'react-native';
import PrimaryButton from '../PrimaryButton';
import SecondaryButton from '../SecondaryButton';

import { styles } from './styles';

export default function FlexButtons(props: FlexButtonsProps) {
    return (
        <View style={[styles.container, props.containerStyle]}>
            <SecondaryButton
                containerStyle={[styles.marginEnd, styles.spanEven]}
                title={props.secondaryButtonText}
                onPress={props.onPressSecondary}
                secondaryTestID={props.secondaryTestID}
                secondaryAccessibilityLabel={props.secondaryAccessibilityLabel}
            />
            <View style={[styles.spanEven]}>
                <PrimaryButton
                    primaryButtonIcon={props.primaryButtonIcon}
                    containerStyle={{ zIndex: 0 }}
                    title={props.primryButtonText}
                    onPress={props.onPressPrimary}
                    primaryTestID={props.primaryTestID}
                    primaryAccessibilityLabel={props.primaryAccessibilityLabel}
                />
                {props.primaryButtonOverlay}
            </View>
        </View>
    );
}

export interface FlexButtonsProps {
    onPressPrimary: () => void;
    onPressSecondary: () => void;
    primryButtonText: string;
    secondaryButtonText: string;
    primaryButtonIcon?: any;
    primaryButtonOverlay?: React.ReactNode;
    containerStyle?: {};
    primaryTestID?: string;
    primaryAccessibilityLabel?: string;
    secondaryTestID?: string;
    secondaryAccessibilityLabel?: string;
}
