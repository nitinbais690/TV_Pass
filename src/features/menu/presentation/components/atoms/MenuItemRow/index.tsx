import React from 'react';
import { View, Text } from 'react-native';
import { MenuItemRowStyles } from './style';
import useAppColors from 'core/presentation/hooks/use-app-colors';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { AUTOMATION_TEST_ID } from 'features/menu/presentation/automation-ids';

/*
   Renders Menu item row
 */
export function MenuItemRow(props: MenuItemProps) {
    const style = MenuItemRowStyles(useAppColors());

    return (
        <TouchableOpacity
            onPress={props.onPress}
            testID={AUTOMATION_TEST_ID.MENU_ITEM + props.title}
            accessibilityLabel={AUTOMATION_TEST_ID.MENU_ITEM + props.title}>
            <View style={style.container}>
                <View style={style.imageAndTextContainer}>
                    {props.image && props.image}
                    <Text style={style.textStyle}> {props.title} </Text>
                </View>
                <View style={style.borderStyle} />
            </View>
        </TouchableOpacity>
    );
}

interface MenuItemProps {
    title: string;
    image: JSX.Element | undefined;
    onPress: () => void;
}
