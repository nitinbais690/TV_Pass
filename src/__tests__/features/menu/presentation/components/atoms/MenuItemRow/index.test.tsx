import { MenuItemRow } from 'features/menu/presentation/components/atoms/MenuItemRow';
import Help from 'assets/images/menu_help.svg';
import React from 'react';
import { render, toJSON } from '@testing-library/react-native';

jest.mock('core/presentation/hooks/use-app-colors');

describe('Menu Item row test', () => {
    it('Menu item render correctly', () => {
        const pressHandler = jest.fn();
        const { container } = render(<MenuItemRow title={'Privacy'} image={<Help />} onPress={pressHandler} />);
        expect(toJSON(container)).toMatchSnapshot();
    });

    it('Menu item render correctly if is not given', () => {
        const pressHandler = jest.fn();
        const { container } = render(<MenuItemRow title={'Privacy'} image={undefined} onPress={pressHandler} />);
        expect(toJSON(container)).toMatchSnapshot();
    });
});
