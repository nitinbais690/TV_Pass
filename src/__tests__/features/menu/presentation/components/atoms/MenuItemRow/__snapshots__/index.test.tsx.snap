// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Item row test Menu item render correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNGestureHandlerButton
    collapsable={false}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityLabel="MENU_ITEM_Privacy"
      accessible={true}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="MENU_ITEM_Privacy"
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "height": 72.85714285714286,
            "justifyContent": "center",
            "paddingHorizontal": 34,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <SvgMock />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "ProximaNova-Semibold",
                "fontSize": 18,
                "fontWeight": "600",
                "paddingLeft": 20,
              }
            }
          >
             
            Privacy
             
          </Text>
        </View>
        <View
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#FFFFFF",
              "opacity": 0.07,
              "paddingTop": 18,
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </RNGestureHandlerButton>
</View>
`;

exports[`Menu Item row test Menu item render correctly if is not given 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNGestureHandlerButton
    collapsable={false}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityLabel="MENU_ITEM_Privacy"
      accessible={true}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="MENU_ITEM_Privacy"
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "height": 72.85714285714286,
            "justifyContent": "center",
            "paddingHorizontal": 34,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "ProximaNova-Semibold",
                "fontSize": 18,
                "fontWeight": "600",
                "paddingLeft": 20,
              }
            }
          >
             
            Privacy
             
          </Text>
        </View>
        <View
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#FFFFFF",
              "opacity": 0.07,
              "paddingTop": 18,
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </RNGestureHandlerButton>
</View>
`;
